x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/details/execution_log_table/execution_log_search_bar.tsx

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { capitalize, replace } from 'lodash';
+import {
+  EuiHealth,
+  EuiFieldSearch,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPopover,
+  EuiFilterGroup,
+  EuiFilterButton,
+  EuiFilterSelectItem,
+} from '@elastic/eui';
+import { RuleExecutionStatus } from '../../../../../../../common/detection_engine/schemas/common';
+import { getStatusColor } from '../../../../../components/rules/rule_execution_status/utils';
+import * as i18n from './translations';
+
+export const EXECUTION_LOG_SCHEMA_MAPPING = {
+  status: 'kibana.alert.rule.execution.status',
+  timestamp: '@timestamp',
+  duration: 'event.duration',
+  message: 'message',
+  gapDuration: 'kibana.alert.rule.execution.metrics.execution_gap_duration_s',
+  indexDuration: 'kibana.alert.rule.execution.metrics.total_indexing_duration_ms',
+  searchDuration: 'kibana.alert.rule.execution.metrics.total_search_duration_ms',
+  totalActions: 'kibana.alert.rule.execution.metrics.number_of_triggered_actions',
+  schedulingDelay: 'kibana.task.schedule_delay',
+};
+
+export const replaceQueryTextAliases = (queryText: string): string => {
+  return Object.entries(EXECUTION_LOG_SCHEMA_MAPPING).reduce<string>(
+    (updatedQuery, [key, value]) => {
+      return replace(updatedQuery, key, value);
+    },
+    queryText
+  );
+};
+
+const statuses = [
+  RuleExecutionStatus.succeeded,
+  RuleExecutionStatus.failed,
+  RuleExecutionStatus['partial failure'],
+];
+
+const statusFilters = statuses.map((status) => ({
+  label: <EuiHealth color={getStatusColor(status)}>{capitalize(status)}</EuiHealth>,
+  selected: false,
+}));
+
+interface ExecutionLogTableSearchProps {
+  onlyShowFilters: true;
+  onSearch: (queryText: string) => void;
+  onStatusFilterChange: (statusFilters: string[]) => void;
+}
+
+export const ExecutionLogSearchBar = React.memo<ExecutionLogTableSearchProps>(
```

Comment:
Can delete if folks want as this isn't currently be used due to missing data issues (but is functional at least).

Reply:
Will add comment why it's currently unused, but will keep for later use

Reply:
Added context in comments in 0efbed2